adam/batch.py

```diff
@@ -388,6 +383,74 @@ def __repr__(self):
     def get_parts(self):
         return self._parts
 
+    def get_final_ephemeris(self):
+        part = self._parts[-1]
+        if part is None:
+            print("Final part is not available")
+            return None
+        state = part.get_calc_state()
+        if (state != 'COMPLETED'):
+            print(
+                "Final part is in state %s, not COMPLETED, so no ephemeris is available" % (state))
+            return None
+        return part.get_ephemeris()
+
+    def _parse_date(self, date):
+        parts = date.split(".")
+        micros = float("0." + parts[1])
+
+        parsed = datetime.strptime(parts[0], "%d %b %Y %H:%M:%S")
+        parsed = parsed + timedelta(microseconds=micros * 1000000)
+        return parsed
+
+    def get_state_vector_at_time(self, target_epoch):
+        """Get the state vector at the given time as a 6d list in [km, km/s]
+
+        This function grabs the STK ephemeris from the final part. It parses the epoch given in
+        the ephemeris in order to determine the times of all the state vectors given in the file.
+
+        If a time is requested that does not have an explicit state vector, None will be returned.
+        This will not interpolate.
```

Comment:
Is interpolating (propagating?) something we want to consider in the future? I could see that being useful.

Reply:
I could see that being useful someday too. If we did support it, I think it'd be something we round-trip to the server for - otherwise we have to figure out either how to make client-side interpolation consistent with the server-side propagator or how to communicate that difference so people don't think they're getting one and are actually getting the other.

Reply:
Yeah, agreed!